data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/stream/ShardAcknowledgementManager.java

```diff
@@ -0,0 +1,231 @@
+package org.opensearch.dataprepper.plugins.source.dynamodb.stream;
+
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSetManager;
+import org.opensearch.dataprepper.model.source.coordinator.enhanced.EnhancedSourceCoordinator;
+
+import org.opensearch.dataprepper.model.source.coordinator.enhanced.EnhancedSourcePartition;
+import org.opensearch.dataprepper.plugins.source.dynamodb.DynamoDBSourceConfig;
+import org.opensearch.dataprepper.plugins.source.dynamodb.coordination.partition.StreamPartition;
+import org.opensearch.dataprepper.plugins.source.dynamodb.coordination.state.StreamProgressState;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import org.opensearch.dataprepper.plugins.source.dynamodb.model.ShardCheckpointStatus;
+
+public class ShardAcknowledgementManager {
+    private static final Logger LOG = LoggerFactory.getLogger(ShardAcknowledgementManager.class);
+
+    private static final String NULL_SEQUENCE_NUMBER = "null";
+
+    private static final long WAIT_FOR_ACKNOWLEDGMENTS_TIMEOUT = 10L;
+
+    static final Duration CHECKPOINT_INTERVAL = Duration.ofSeconds(10);
+
+    private final DynamoDBSourceConfig dynamoDBSourceConfig;
+    private final Map<StreamPartition, ConcurrentLinkedQueue<ShardCheckpointStatus>> checkpoints = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<StreamPartition, ConcurrentHashMap<String, ShardCheckpointStatus>> ackStatus = new ConcurrentHashMap<>();
```

Comment:
Rename this for clarity. Looking at some of the code below, it looked like a local variable. Maybe `ackStatuses` would help clarify.